### PR TITLE
Upgrade to tchannel-2.0.0-beta6

### PR DESCRIPTION
**Pull request by jcorbin**
cc @jwolski @Raynos

### CODE REVIEW

**Comment by jcorbin** [comment]
Ready for review @jwolski @Raynos

File: lib/request-proxy/send.js [hunk 1]
```diff
@@ -223,19 +223,35 @@ RequestProxySend.prototype.scheduleRetry = function scheduleRetry(callback) {
...
         this.request.body,
-        this.maxRetries === 0 ? callback : onSend
+        onSend
     );
 
     this.ringpop.emit('requestProxy.requestProxied');
 
-    function onSend(err, res1, res2) {
+    function onSend(err, res, arg2, arg3) {
+        if (!err && res && !res.ok) {
```
Comment (jwolski) [inline lib/request-proxy/send.js]: here we check whether `res` actually exists. Other places, we don't. What is true?

Comment (Raynos) [inline lib/request-proxy/send.js]: `res` always exist if `err` is `null`

Comment (jwolski) [inline lib/request-proxy/send.js]: thanks jakebro

**Comment by Raynos** [comment]
I'm going to rebase on master & squash this branch. Wish me good luck.

**Comment by jwolski** [comment]
I've already tried merging master into this branch one last time. There are some broken tests that I didn't have time to fix and 1 conflict that I experience. Let me know if you need help with the conflict.

**Comment by Raynos** [comment]
I resolved conflicts & am fixing tests

**Pull request closed without merging**